### PR TITLE
Update documentation about downloading CodeQL databases using the VS Code extension

### PR DESCRIPTION
**Pull request by jf205**
See [internal issue](https://github.com/github/docs-content/issues/7800).

### CODE REVIEW

File: docs/codeql/reusables/download-github-database.rst
Comment (felicitymay) [inline docs/codeql/reusables/download-github-database.rst:2]: I think we're going to want to link out to the new download CodeQL database endpoint here, once the OpenAPI changes are published. Presumably that content will explain what the acceptable values for `<language>` are?

It may be worth giving a little more information about the endpoint here, since the GitHub CLI isn't the only way to download a database from GitHub.com. Then just use the GH CLI example below as an example of a way to use the endpoint.

File: docs/codeql/codeql-cli/getting-started-with-the-codeql-cli.rst
Comment (felicitymay) [inline docs/codeql/codeql-cli/getting-started-with-the-codeql-cli.rst:52]: I'm wondering if we want to mention that if you want to use the CLI to generate code scanning data for upload to GitHub Enterprise server, then you should use the recommended version? We could link to: [Downloading the CodeQL CLI](https://docs.github.com/en/enterprise-server@3.6/code-security/code-scanning/using-codeql-code-scanning-with-your-existing-ci-system/installing-codeql-cli-in-your-ci-system#downloading-the-codeql-cli) (which I see has a typo 🤦🏻‍♀️).

File: docs/codeql/codeql-for-visual-studio-code/analyzing-your-projects.rst
Comment (felicitymay) [inline docs/codeql/codeql-for-visual-studio-code/analyzing-your-projects.rst:17]: > from a project slug on GitHub.com

Do we describe how to do this somewhere? It doesn't seem to be mentioned in the section on getting a database from GitHub.com.

File: docs/codeql/reusables/download-github-database.rst
Comment (felicitymay) [inline docs/codeql/reusables/download-github-database.rst:4]: Is there any way to find out whether or not a CodeQL database is available for a project, either in the GitHub UI or via a REST API call?

**Review by felicitymay — COMMENTED** [review]
Many thanks for updating the docs for this upcoming release ✨ 🙇🏻‍♀️ 

I've added a few suggestions, but generally this looks good to me.

File: docs/codeql/codeql-for-visual-studio-code/analyzing-your-projects.rst
Comment (jf205) [reply docs/codeql/codeql-for-visual-studio-code/analyzing-your-projects.rst]: I've changed this to "project's URL on GitHub.com" -- the UI is quite clear, so i think that should be sufficient. 

Screenshot 2022-09-13 at 14 31 47

**Comment by jf205** [comment]
Thanks for the first review @felicitymay ⚡ . I think this is ready for another 👀  when you have time please!

File: docs/codeql/codeql-cli/creating-codeql-databases.rst
Comment (felicitymay) [reply docs/codeql/codeql-cli/creating-codeql-databases.rst:392]: Thank you 💖

File: docs/codeql/reusables/download-github-database.rst
Comment (felicitymay) [inline docs/codeql/reusables/download-github-database.rst:3]:
```suggestion
You can check if a repository has any CodeQL databases available for download using the ``/repos/<owner>/<repo>/code-scanning/codeql/databases`` endpoint.
```

File: docs/codeql/reusables/download-github-database.rst
Comment (felicitymay) [inline docs/codeql/reusables/download-github-database.rst:8]:
```suggestion
This command returns information about any CodeQL databases that are available for a repository, including the language the database represents, and when the database was last updated. If no CodeQL databases are available, the response is empty.
```

File: docs/codeql/codeql-cli/getting-started-with-the-codeql-cli.rst
Comment (felicitymay) [inline docs/codeql/codeql-cli/getting-started-with-the-codeql-cli.rst:54]: My apologies. I've just remembered that we have a version of this link that always shows the latest GHES article.

```suggestion
   * If you want to generate code scanning data to upload to GitHub Enterprise server, then download the version that is compatible with the CodeQL CLI used in your CI system. For more information, see "`Installing CodeQL CLI in your CI system <https://docs.github.com/en/enterprise-server/code-security/code-scanning/using-codeql-code-scanning-with-your-existing-ci-system/installing-codeql-cli-in-your-ci-system#downloading-the-codeql-cli>`__" in the GitHub documentation.
```

**Review by felicitymay — COMMENTED** [review]
Thanks for the updates. Just a couple of small loose ends, otherwise LGTM once we have the REST API URL. 💖

**Comment by jf205** [comment]
Thanks for the reviews @felicitymay. I've accepted your suggestions and fixed the remaining TODO. We can merge it when the accompanying changes are ready.

**Review by felicitymay — APPROVED** [review]

**Review by aeisenberg — APPROVED** [review]

**Comment by jf205** [comment]
Apologies for the broad review request -- i messed up while changing target branch 🤦🏻

**Comment by jf205** [comment]
@felicitymay and @aeisenberg: unfortunately i accidentally dismissed your approvals when i tried (and failed) to retarget this to rc/3.7. Please can i get another approval? Then i'll get this merged and port the changes over in another PR

**Review by aeisenberg — APPROVED** [review]

**Comment by aeisenberg** [comment]
@jf205 let me know if you need some help re-targeting this for the rc/3.7 branch.

**Pull request merged**
